Cabal/Distribution/Backpack/LinkedComponent.hs

```diff
@@ -205,7 +212,13 @@ toLinkedComponent verbosity db this_pid pkg_map ConfiguredComponent {
                           modScopeProvides =
                             Map.unionWith (++)
                               local_exports
-                              (modScopeProvides linked_shape0)
+                              (modScopeProvides linked_shape0),
+                          -- TODO: test that requirements aren't already
+                          -- in scope
+                          modScopeRequires =
+                            Map.unionWith (++)
```

Comment:
This one isn't a change in behavior: previously the local requirements were linked in during mixin linking so they were already in linked_shape0 ;)